### DISCUSSION TITLE
Where is the source code for AppWindow/Window?

### DISCUSSION THREAD

**Discussion by whiskhub** [microsoft/microsoft-ui-xaml#9905] [Q&A] [answered]
Hi everyone, 

I was trying to see the C++ source code for the [Window](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.window?view=windows-app-sdk-1.5) and [AppWindow](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.windowing.appwindow?view=windows-app-sdk-1.5) classes, but am not able to find it.

I've done searches across both [microsoft-ui-xaml](https://github.com/microsoft/microsoft-ui-xaml) and [WindowsAppSDK](https://github.com/microsoft/WindowsAppSDK) repos, first using GitHub code search and even then cloning the repo locally. For example for something concrete like the `IsShownInSwitchers` property. But no results.

Could someone provide a link to the source files, and how you found them?

Thanks!

**Comment by castorix** [comment]
I did not find the exact source code, but _IsShownInSwitchers_ calls ITaskbarList::HrInit + ITaskbarList::DeleteTab

**Comment by DarranRowe** [comment]
There doesn't seem to be anything for AppWindow, but the partial source for Window should mostly be in the directory src\dxaml\xcp\dxaml\lib.

**Comment by codendone** [comment, marked as answer]
Window is primarily a combination of [Window_Partial.cpp](https://github.com/microsoft/microsoft-ui-xaml/blob/winui3/release/1.5-stable/dxaml/xcp/dxaml/lib/Window_Partial.cpp), which is the direct Window APIs, and [DesktopWindowImpl.cpp](https://github.com/microsoft/microsoft-ui-xaml/blob/winui3/release/1.5-stable/dxaml/xcp/dxaml/lib/DesktopWindowImpl.cpp), which provides the underlying implementation.

AppWindow is defined and implemented in a lower layer of code which is not shared.
